### ISSUE TITLE
[BUG] Literal True query filter gives unexpected value error

### ISSUE THREAD

**Issue by Andrioden** [vitalik/django-ninja#905]
**Describe the bug**

Using `Optional[Literal[True]]` Schema type gives `HTTP 422 unexpected value`, se stripped example below.

api
``` python
@router.get("/", response=...)
def get_many(request: WSGIRequest, filters: PlayersFilters = Query(...)) -> ...:
    # ...
```

model
``` python
class PlayersFilters(Schema):
    discovered: Optional[Literal[True]]
```

Test api calls
```
/api/players/?discovered=true
/api/players/?discovered=True
```

Error
```
HTTP 422
content = b'{"detail": [{"loc": ["query", "discovered"], "msg": "unexpected value; permitted: True", "type": "value_error.const", "ctx": {"given": "true", "permitted": [true]}}]}'
```

**Versions (please complete the following information):**
 - Python version: 3.11
 - Django version: 4.2.6
 - Django-Ninja version: 0.22.2
 - Pydantic version: 1.10.13

**Comment by Andrioden** [comment]
Calling `PlayersFilters(discovered=True)` works, so pydantic is not the problem

**Comment by vitalik** [comment]
@Andrioden well the thing is that from GET you are getting strings as values

so basically you are getting this to execute:
```Python
PlayersFilters(discovered='true')
```
which is not validated

maybe what you need is actually `discovered: bool`  (bool type automatically converts string 'true' to boolean)

**Comment by Andrioden** [comment]
Yeah, i assumed it was something like that. I changed my code to 

``` python
class PlayersFilters(Schema):
    discovered: Optional[bool]
```

and that works. But would be nice if `Optional[Literal[True]]` worked! Maybe I can add some pydantic Field filters to do the same thing thou, didnt try that.

**Comment by vitalik** [comment]
well with pydantic2 you can add some pre-validator that will turn string 'true' to bool-True and then your Literal will work

**Comment by Andrioden** [comment]
I can check that out when i upgrade to django ninja 1 and pydantic 2. I assume this is to say, you cant easily and have no goal of adding support/fix for this out of the box? I would think its a goatcha other could experience as well.
